pkg/command/root.go

```diff
@@ -240,8 +247,41 @@ func newRootCmd() *cobra.Command {
 		// silencing usage for now as we are getting double usage from plugins on errors
 		SilenceUsage: true,
 		PersistentPreRunE: func(cmd *cobra.Command, args []string) error {
-			// Sets the verbosity of the logger if TANZU_CLI_LOG_LEVEL is set
-			setLoggerVerbosity()
+			// Manually parse the flags
+			for i := 0; i < len(args); i++ {
+				arg := args[i]
+				if arg == "-v" || arg == "--verbose" {
+					if i+1 < len(args) {
+						nextArg := args[i+1]
+						// Check if the next argument is another flag
+						if strings.HasPrefix(nextArg, "-") {
+							log.Errorf("Missing value for verbose flag")
+						}
+						// Try to convert the next argument to an integer
+						if v, err := strconv.Atoi(nextArg); err == nil {
+							verbose = v
+						} else {
+							log.Errorf("Invalid value for verbose flag: %v\n", err)
+						}
+						// Skip the next argument
+						i++
+					} else {
+						log.Errorf("Missing value for verbose flag")
+					}
+				}
+			}
+
+			// Sets the verbosity of the logger
+			setLoggerVerbosity(verbose)
```

Comment:
Should we fail execution if the user sets an invalid value for `--verbose`?
Notice the difference:
```
$ tz version --verbose allo
[x] : invalid argument "allo" for "--verbose" flag: strconv.ParseInt: parsing "allo": invalid syntax

$ tz version --verbose -10
[i] I am level default
[i] I am level 1
[i] I am level 2
[i] I am level 3
...
```

Reply:
Added validation to verbose flag
`> tz version --verbose -10
Error: Invalid value for verbose flag. It should be equal or greater than 0
> tz telemetry --verbose -10
[x] Missing value for verbose flag
`